GoogleApiWrappers/src/Google.Apis.Storage.v1.ClientWrapper/StorageClient.DownloadObject.cs

```diff
@@ -0,0 +1,165 @@
+﻿// Copyright 2015 Google Inc. All Rights Reserved.
+// Licensed under the Apache License Version 2.0.
+
+using Google.Apis.Download;
+using Google.Common;
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Object = Google.Apis.Storage.v1.Data.Object;
+
+namespace Google.Apis.Storage.v1.ClientWrapper
+{
+    public partial class StorageClient
+    {
+        // Note: documentation will be provided when we've worked out if we really want 16 methods here.
+        // See issue #21.
+
+        public void DownloadObject(string bucket, string sourceName, Stream destination)
```

Comment:
`objectName` might be best then - I think it *is* important to have `name` in there, as that's where you'd have got the value from. In other words, these two calls are broadly equivalent:

```
Object obj = ...;
client.DownloadObject(obj.Bucket, obj.Name, ...);
client.DownloadObject(obj);
```